tensorbay/cli/branch.py

```diff
@@ -23,6 +25,10 @@ def _implement_branch(obj: Dict[str, str], tbrn: str, name: str, verbose: bool)
     gas = get_gas(**obj)
     dataset_client = get_dataset_client(gas, info)
 
+    if is_delete:
+        _delete_branch(dataset_client, name)
```

Comment:
The name of the branch to be deleted will be in the `tbrn` rather than the `name`.